app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+        val rView = view.findViewById<ForceEditText>(R.id.rgbR)
+        val gView = view.findViewById<ForceEditText>(R.id.rgbG)
+        val bView = view.findViewById<ForceEditText>(R.id.rgbB)
+        val chView = view.findViewById<ForceEditText>(R.id.colorHex)
+        val slider = view.findViewById<ColorSlider>(R.id.slider)
+        val square = view.findViewById<GradientSquare>(R.id.gradientSquare)
+        val activeBox = view.findViewById<TextView>(R.id.colorOne)
+        val previousBox = view.findViewById<TextView>(R.id.colorTwo)
+        val previousColor = onColorTextForceUpdate().previousColor
+        previousBox.setBackgroundColor(previousColor.toArgb())
+        //used to prevent dragging when clicking on the gradient or the slider
+        val onViewTouch = {
+            sheet.behavior.isDraggable = false
+        }
+        val onViewLift = {
+            sheet.behavior.isDraggable = true
+        }
+        //used to set the color of the active box
+        val setActiveBox: (FloatArray) -> Unit = { hsv ->
+            val intColor = Color.HSVToColor(hsv)
+            val red = Color.red(intColor)
+            val green = Color.green(intColor)
+            val blue = Color.blue(intColor)
+
+            hView.forceText(hsv[0].toString())
+            sView.forceText((hsv[1] * 100).toString())
+            vView.forceText((hsv[2] * 100).toString())
+            rView.forceText(red.toString())
+            gView.forceText(green.toString())
+            bView.forceText(blue.toString())
+            chView.forceText(Integer.toHexString(Color.rgb(red, green, blue)).drop(2))
+
+            square.hue = hsv[0]
+            square.sat = hsv[1]
+            square.valu = hsv[2]
+            square.invalidate()
+
+            slider.sliderSetHue(hsv[0])
+
+            activeBox.setBackgroundColor(intColor)
+        }
+        //called whenever text needs to be updated
+        val updateForceViews = {
+            val pack = onColorTextForceUpdate()
+            setActiveBox(floatArrayOf(pack.hue, pack.saturation, pack.value))
+        }
+        previousBox.setOnClickListener {
+            val hsv = FloatArray(3)
+            Color.colorToHSV(previousColor.toArgb(), hsv)
+            //used to set the color of the active box to the previous color
+            setActiveBox(hsv)
+        }
+        submit.setOnClickListener {
+            onSubmission(
+                Color.HSVToColor(
+                    floatArrayOf(
+                        slider.hue,
+                        square.sat,
+                        square.valu
+                    )
+                ).toColor()
+            )
+            sheet.dismiss()
+        }
+        cancel.setOnClickListener {
+            sheet.dismiss()
+        }
+
+        slider.onHueChange = {
+            square.hue = it
+            square.invalidate()
+            onHueUpdate(it)
+            updateForceViews()
+        }
+        square.onSatOrValChange = { saturation, value ->
```

Comment:
your going to want to release the focus you have on your editTexts here, if they are in focus they won't update properly from what I encountered.

Reply:
It already does so.